.github/workflows/commit-checker.yml

```diff
@@ -0,0 +1,42 @@
+name: Commit checker
+
+on:
+  pull_request:
+
+jobs:
+  commit-checker:
+    name: Commit checker
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 3
+
+    - name: Get pull-request commits
+      run: |
+        set -x
+        # We are in a merge, HEAD^ is the base branch, HEAD^2 is the merged branch
+        # Get the latest commit from the pull-request in history. If no parent is found for the
+        # latest commit it may not be the latest, increase history
+        while [ ! $(git rev-parse --verify `git rev-list --reverse HEAD^..HEAD^2 | head -n1`^) ]; do
+            git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --deepen=3 origin HEAD
```

Comment:
```
DEPTH=4
...
while ...
--deepen=${DEPTH}
DEPTH=$(( ${DEPTH} * 4 ))
```

So that means first 4 more commits, 16, 64, ...

Worst case it will be fetching 24000 commits.  So that would be around 6 calls. Not pretty, but .. I guess there is a balance to be found.